nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/trash/TrashService.java

```diff
@@ -167,7 +178,9 @@ default void untrashDocument(DocumentModel doc) {
      * @param parent The parent document of trash document.
      * @return All documents in the trash of the current document.
      * @since 7.1
+     * @deprecated since 11.1 not scalable (prefer to use directly {@link CoreSession#query(String)} with a limit
      */
```

Comment:
here you don't give enough informations, which method will replace the deprecated once, ` CoreSession#query(String)` with a limit its not enough because perhaps the called want the total document without using limit. Perhaps just `@deprecated since 11.1, not scalable or not used anymore` or at least you can propose to replace it by `BAF`